core/src/main/java/kafka/server/builders/KafkaApisBuilder.java

```diff
@@ -179,7 +179,7 @@ public KafkaApis build() {
         if (metrics == null) throw new RuntimeException("You must set metrics");
         if (quotas == null) throw new RuntimeException("You must set quotas");
         if (fetchManager == null) throw new RuntimeException("You must set fetchManager");
-        if (brokerTopicStats == null) brokerTopicStats = new BrokerTopicStats(Optional.of(config));
+        if (brokerTopicStats == null) brokerTopicStats = new BrokerTopicStats(config.remoteLogManagerConfig().enableRemoteStorageSystem());
```

Comment:
nit: I can see another def `isRemoteLogStorageSystemEnabled` exposed directly in KafkaConfig for same hence we might want to use that directly.

```suggestion
        if (brokerTopicStats == null) brokerTopicStats = new BrokerTopicStats(config.isRemoteLogStorageSystemEnabled());
```

Reply:
Yes we can use that. Thanks for the suggestion!

I see we both methods in different files, it would probably make sense to stick to one and be consistent. I'll keep this PR focused on removing the KafkaConfig dependency, we can follow up with another PR to do the cleanup.

Reply:
I created https://issues.apache.org/jira/browse/KAFKA-16859 to do the cleanup.

Reply:
Thanks 👍